code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java

```diff
@@ -149,4 +161,71 @@ public void buildParentWebHookWhenIsHookAvailable() throws IOException {
         verify(gitConfig, times(1)).getUser();
         verify(gitConfig, times(1)).getParentRepositoryName();
     }
+
+    @Nested
+    @ExtendWith(MockitoExtension.class)
+    class DeleteTests {
+
+        private final String REPO_PREFIX = "gamified-hiring-";
+        private final String GAME = "kata";
+        private final String SUFFIX = "-" + GAME;
+        private final String RANDOM_REPO_NAME = "some random name: ";
+
+        private Map<String, GHRepository> allRepos;
+        private Map<String, GHRepository> reposForDeletion;
+        private Map<String, GHRepository> reposToKeep;
+
+        @BeforeEach
+        public void setUp() throws Exception{
+            allRepos = new HashMap<>();
+            reposForDeletion = new HashMap<>();
+            reposToKeep = new HashMap<>();
+
+            for (int i = 0; i < 100; i++) {
+                GHRepository repo = mock(GHRepository.class);
+                when(repo.getName()).thenReturn(REPO_PREFIX+i+SUFFIX);
+                allRepos.put(repo.getName(), repo);
+                reposForDeletion.put(repo.getName(), repo);
+            }
+
+            for (int i = 0; i < 100; i++) {
+                GHRepository repo = mock(GHRepository.class);
+                when(repo.getName()).thenReturn(RANDOM_REPO_NAME + i);
+                allRepos.put(repo.getName(), repo);
+                reposToKeep.put(repo.getName(), repo);
+            }
+
+            when(gitHub.getMyself()).thenReturn(myself);
+            when(myself.getAllRepositories()).thenReturn(allRepos);
+        }
+
+        @org.junit.jupiter.api.Test
```

Comment:
why is this imported differntly

Reply:
@Test from junit4 doesn't work with @nested test class, so @Test from JunitJupiter had to be used